src/t_stream.c

```diff
@@ -3354,6 +3598,46 @@ void xinfoReplyWithStreamInfo(client *c, stream *s) {
                 addReplyBulkCString(c,"last-delivered-id");
                 addReplyStreamID(c,&cg->last_id);
 
+                /* Offset of the last delivered ID */
+                addReplyBulkCString(c,"last-delivered-offset");
+                addReplyLongLong(c,cg->offset);
+
+                /* Group lag */
+                int valid = 1;
+                uint64_t lag = 0;
+
+                addReplyBulkCString(c,"lag");
+                if (!s->offset) {
+                    /* The lag of a newly-initialized stream is 0. */
+                    lag = 0;
+                } else if (cg->offset && streamIsContiguousRange(s,&cg->last_id)) {
+                    /* No fragmentation ahead means that the group's
+                        * offset is valid for lag calculation. */
+                    lag = s->offset - cg->offset;
+                } else if (streamIDEqZero(&cg->last_id)) {
+                    if (streamIsContiguousRange(s,NULL)) {
+                        /* The group is at 0-0 of a non-fragmented stream. */
+                        lag = s->length;
+                    } else {
+                        valid = 0;
+                    }
+                } else {
+                    /* Attempt to retrieve the group's pffset. */
+                    uint64_t offset = streamGetOffsetForTip(s,&cg->last_id);
+                    if (offset) {
+                        /* A valid offset was obtained - w00t! */
+                        lag = s->offset - offset;
+                    } else {
+                        valid = 0;
+                    }
+                }
```

Comment:
is that the only way to get the lag? is it only an introspection feature?
maybe we want to add some XLAG command?

Reply:
Yep, that's the only way to get it. Yep, it's only for introspection/monitoring (potentially triggering a client-side scaling event).

An XLAG command would be sugar-syntax for `XINFO`, so I'm not sure the functionality deserves the API bloat.